### ISSUE TITLE
Login/Logout Redirect issue with multilingual Site with SEF

### ISSUE THREAD

**Issue by asfaque** [joomla/joomla-cms#21448]
### Steps to reproduce the issue
1) Add to different language to the Joomla site (In my case, I had Arabic and English)
2) Creating a menu and select Users -> login

### Expected result
It Should redirect me the home page or page i have selected in menu.

### Actual result
It is giving an error while logging or if i have logged in by turning SEF Urls off.

Even tried on Joomla 3.8.8 to 3.8.11, Protostar templates

### Additional comments

![screen shot 2018-08-07 at 13 37 05](https://issues.joomla.org/uploads/1/4e57c8ff359b18919928d20cb21986e8.png)![screen shot 2018-08-07 at 13 37 06](https://issues.joomla.org/uploads/1/ad6a8c10cf91f36a91ba53a1e835294f.png)

**Comment by ghost** [comment]
do you mean in Title "... Site with SEF off"?

**Comment by asfaque** [comment]
No when SEF urls in global configuration is set to **yes**, I got above error and when i set it to **No** it is working fine. The Problem is when SEF urls is set to **NO**, In urls we got in the ?lang=en but when we set SEF to **yes** the urls we got index.php/en/alias. And I think this is the problem.

**Comment by ghost** [comment]
thanks for Answer, @asfaque. 

@infograf768?

**Comment by infograf768** [comment]
SEF on or Off. Can't reproduce.
Created menu items of type `Login Form` in each language specific Main Menu.
When logging through these menu items, I have absolutely no error.

Are you using URL Rewriting? Is the .htaccess file existing?

**Comment by ghost** [comment]
thanks @infograf768.

@asfaque please ask help on the [forums](https://forum.joomla.org/). This repository concerns in first Place Joomla-Core coding. For this Reason closing this Issue, thanks.

**Comment by asfaque** [comment]
@franz-wohlkoenig @infograf768 Its a core issue, it would be great if you guys had understand the issue before closing the issue.

And to reproduce, 
1) Install Arabic language, with english as default
2) Enable Content language.
3) Create Three menu: Main Menu, Arabic Menu and Other/Hidden
4) In Main and Arabic Menu create menu item, set is language to their respective one and make them default as well.
5) To Make multilingual work, we need to create one more menu item let say login in other/hidden menu and make it default with **All Language**.
6) Enable language Filter and Code system plugin.
7) Enable Language Switcher Module in Position 7 of protostar.
8) Now try login from frontend.


And please for god sake try it one more time and its a core issue.
![screen shot 2018-08-08 at 06 19 52](https://issues.joomla.org/uploads/1/08566b5a2ff72557fe10fed782cfd20f.png)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21448">issues.joomla.org/tracker/joomla-cms/21448</a>.</sub>

**Comment by ghost** [comment]
reopened as stated above.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21448">issues.joomla.org/tracker/joomla-cms/21448</a>.</sub>

**Comment by ghost** [comment]
> Are you using URL Rewriting? Is the .htaccess file existing?

@asfaque can you please answer above Questions by @infograf768?

**Comment by asfaque** [comment]
No I'm Working locally, .htaccess doesn't work on my local.
And even i'm not using URL Rewriting.
---

Now i discover that login module is working completely fine. This problem is only with the com_users<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21448">issues.joomla.org/tracker/joomla-cms/21448</a>.</sub>

**Comment by ggppdk** [comment]
@infograf768 

> SEF on or Off. Can't reproduce.
> Created menu items of type Login Form in each language specific Main Menu.
> When logging through these menu items, I have absolutely no error.

it is tricky
yes it will work
because you are using **same component** (and same view) for each home page menu items 

but if your language specific home page menu items are made to point to **different components**
than the home page of language all , then you will get the issue described / explained here:

https://github.com/joomla/joomla-cms/issues/20074

Please note **that i have not tested steps of this issue**,
but it looks like the issue i link above (as far as i can remember and looking at my issue again)

also even if you use same component but different views, you may get issue too (please note that i have not tested this case)

in short the logic of router for home page menu items, assumes that all of them are the same / component view

@asfaque 
please test this
make all home page menu item, to be of the exact same menu item type 
does it work ?
if it does not work then this must be a different issue

**Comment by asfaque** [comment]
@ggppdk I m using the same item type for all menu item. It is a pagebuilder component and even tried on featured article type as well.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21448">issues.joomla.org/tracker/joomla-cms/21448</a>.</sub>

**Comment by ggppdk** [comment]
> I m using the same item type for all menu item. It is a pagebuilder component and even tried on featured article type as well.

Ok then, it must be a different issue then

just the picture that you posted

shows the home menus items to be pagebuilder component 
-- and the 'ALL' home page menu item to be of type "login form"

can you confirm that also home page menu item for 'ALL' is same type as the others ?

**Comment by asfaque** [comment]
yes.

1) Two home page menu item as default with their respective language in different menus.
2) One Login with Language set to **ALL** in different menu let say hidden/other.

Check the instruction above as well https://issues.joomla.org/tracker/joomla-cms/21448#event-379956<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21448">issues.joomla.org/tracker/joomla-cms/21448</a>.</sub>

**Comment by ggppdk** [comment]
@asfaque 

I mean to click on the home page menu item for language all
and then change (/check) its type (parameter "**Menu Item Type**")
so that it is same type as the other home page menu items,
aka be of type 'Page builder'

Please look at the picture that you posted,
it seems to me that home page menu item for language 'ALL' is of different type

![diff_home_menu_all](https://user-images.githubusercontent.com/5092940/43821843-e1723b6c-9af2-11e8-96ad-933bf7f4ee39.png)

**Comment by asfaque** [comment]
Yes because we have to make one default menu item which has language set to all<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21448">issues.joomla.org/tracker/joomla-cms/21448</a>.</sub>

**Comment by asfaque** [comment]
Alright what i did is, i created a new menu item home with same menu type and set its language to all. And it is working<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21448">issues.joomla.org/tracker/joomla-cms/21448</a>.</sub>

**Comment by asfaque** [comment]
But by doing i have to make three home pages, Aren't this gonna be a problem on google seo.
Because i can't create menu item alias for **ALL** default menu and have to select same page with two menus.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21448">issues.joomla.org/tracker/joomla-cms/21448</a>.</sub>

**Comment by asfaque** [comment]
But by doing i have to make three home pages, Aren't this gonna be a problem on google seo.
Because i can't create menu item alias for **ALL** default menu and have to select same page with two menus.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21448">issues.joomla.org/tracker/joomla-cms/21448</a>.</sub>

**Comment by ghost** [comment]
@asfaque please ask help on the [forums](https://forum.joomla.org/). This repository concerns in first Place Joomla-Core coding. For this Reason closing this Issue, thanks.

**Comment by infograf768** [comment]
Basic of any Multilingual site is to have a specific menu with a home set to All languages and never publish the menu module for that menu.

Then you have to make a specific menu per Content Language, each containing a home menu item set to that language and their menu module set to the same content language.

Any login form menu item created has OR to be set to ALL languages and added in EACH Content Language Menu containing the specific Home page OR set tot he same language as the Home in these menus, OR added to another menu which menu module is set to ALL.
